lib/ui/platform_dispatcher.dart

```diff
@@ -2605,3 +2662,52 @@ class SemanticsActionEvent {
     );
   }
 }
+
+/// Signature for [PlatformDispatcher.onPlatformFocusChange].
+typedef ViewFocusChangeCallback = void Function(ViewFocusEvent);
+
+/// An event for the engine to communicate view focus changes back to the app.
+final class ViewFocusEvent {
+  /// Creates a [ViewFocusChange].
+  const ViewFocusEvent({
+    required this.viewId,
+    required this.focusState,
+  });
+
+  /// The id of the [FlutterView] that experienced a focus change.
+  final int viewId;
+
+  /// The current [ViewFocusState] of the [FlutterView] with the given [viewId].
+  final ViewFocusState viewFocusState;
+}
+
+/// Represents a given flutter view focus state.
+///
+/// If focus was lost, use [ViewFocusState.unfocused].
+/// If focus was gained without direction, use [ViewFocusState.focused].
+/// If focus was gained in forward direction, use [ViewFocusState.forwardFocused].
+/// If focus was gained in backwards direction, use [ViewFocusState.backwardsFocused].
+///
+/// Valid transitions, within a view, are:
+///
+/// - [ViewFocusState.unfocused] to [ViewFocusState.focused].
+/// - [ViewFocusState.unfocused] to [ViewFocusState.forwardFocused].
+/// - [ViewFocusState.unfocused] to [ViewFocusState.backwardsFocused].
+/// - [ViewFocusState.focused] to [ViewFocusState.unfocused].
+/// - [ViewFocusState.forwardFocused] to [ViewFocusState.unfocused].
+/// - [ViewFocusState.backwardsFocused] to [ViewFocusState.unfocused].
+///
+/// All others are invalid.
+enum ViewFocusState {
+  /// Used to indicate a given [FlutterView] lost focus.
+  unfocused,
+
+  /// Used to indicate a given [FlutterView] gained focus without direction.
+  focused,
```

Comment:
One thing I have been wondering: How should apps respond to this directionless event? Should the thing that last had focus in the view be refocused? The first thing? A random thing?

Reply:
If we separate the enum into two values, like you are suggesting below, then the specification of what exactly needs to happen when a view gains focus can be supplied by the second value. In my reply, I'm also suggesting how we could structure the second value for future extensibility, if we expect that we won't be able to enumerate all possible responses the framework would need to implement to a focus request.

Reply:
I'm actually leaning towards the idea of removing `focused` (without direction). At least in web `forward` focus seems to be assumed unless specified otherwise.

Reply:
We can always add `non directional` focus later if we find a concrete use case.

Reply:
Done